var/spack/repos/builtin/packages/rocblas/package.py

```diff
@@ -74,29 +75,35 @@ def check(self):
     depends_on('hip@4.1.0:', when='@4.1.0:')
     depends_on('llvm-amdgpu@4.1.0:', type='build', when='@4.1.0:')
     depends_on('rocm-cmake@master', type='build', when='@master:')
+    depends_on('rocm-cmake@5.2.0:', type='build', when='@5.2.0:')
     depends_on('rocm-cmake@4.5.0:', type='build', when='@4.5.0:')
     depends_on('rocm-cmake@4.3.0:', type='build', when='@4.3.0:')
     depends_on('rocm-cmake@3.5.0:', type='build')
 
     for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0', '4.1.0',
                 '4.2.0', '4.3.0', '4.3.1', '4.5.0', '4.5.2', '5.0.0', '5.0.2',
-                '5.1.0', '5.1.3']:
+                '5.1.0', '5.1.3', '5.2.0']:
         depends_on('hip@' + ver,                         when='@' + ver)
         depends_on('llvm-amdgpu@' + ver,  type='build',  when='@' + ver)
         depends_on('rocminfo@' + ver,     type='build',  when='@' + ver)
 
-    depends_on('python@3.6:', type='build')
+    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0']:
+        depends_on('rocm-smi@' + ver, type='build', when='@' + ver)
+
+    for ver in ['4.0.0', '4.1.0', '4.2.0', '4.3.0', '4.3.1', '4.5.0', '4.5.2',
+                '5.0.0', '5.0.2', '5.1.0', '5.1.3']:
+        depends_on('rocm-smi-lib@' + ver, type='build', when='@' + ver)
 
-    with when('+tensile'):
-        # default library format since 3.7.0
-        depends_on('msgpack-c@3:', when='@3.7:')
+    # This is the default library format since 3.7.0
+    depends_on('msgpack-c@3:', when='@3.7:')
 
-        depends_on('py-virtualenv', type='build')
-        depends_on('perl-file-which', type='build')
-        depends_on('py-pyyaml', type='build')
-        depends_on('py-wheel', type='build')
-        depends_on('py-msgpack', type='build')
-        depends_on('py-pip', type='build')
+    depends_on('python@3.6:', type='build')
+    depends_on('py-virtualenv', type='build')
+    depends_on('perl-file-which', type='build')
+    depends_on('py-pyyaml', type='build')
+    depends_on('py-wheel', type='build')
+    depends_on('py-msgpack', type='build')
+    depends_on('py-pip', type='build')
```

Comment:
This appears to have gotten a bit mixed up during merge conflict resolution, as you're now undoing #31414. There should be not be any changes to this section in the diff for this PR.

Reply:
Corrected the merge conflict.